### PR TITLE
Policy: Import kubernetes UID into the policy.

### PR DESCRIPTION
**Pull request by eloycoto**
Kubernetes creates a new UID per each policy, with this change the UID
is saved in the policy labels and can be used for referencing policies
for future work.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6138)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: pkg/k8s/network_policy.go [hunk 1]
```diff
@@ -49,15 +48,16 @@ func GetPolicyLabelsv1(np *networkingv1.NetworkPolicy) labels.LabelArray {
 		return nil
 	}
 
-	policyName := np.Annotations[annotation.Name]
+	policyName := np.Name
```
Comment (aanm) [inline pkg/k8s/network_policy.go]: Revert this change and make it in a new PR .

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/k8s/network_policy.go (hunk 1 above)
Comment (eloycoto) [reply pkg/k8s/network_policy.go]: Done.

**Comment by eloycoto** [comment]
test-me-please

File: pkg/k8s/network_policy_test.go [hunk 2]
```diff
@@ -1625,11 +1636,11 @@ func (s *K8sSuite) TestGetPolicyLabelsv1(c *C) {
 		lbls := GetPolicyLabelsv1(tt.np)
 
 		c.Assert(lbls, NotNil)
-		c.Assert(len(lbls), Equals, 3, Commentf("Incorrect number of labels: Expected Name, Namespace and DerivedFrom labels."))
-
+		c.Assert(len(lbls), Equals, 4, Commentf("Incorrect number of labels: Expected Name, Namespace and DerivedFrom labels."))
```
Comment (aanm) [inline pkg/k8s/network_policy_test.go]: please update this comment

File: pkg/k8s/network_policy_test.go [hunk 3]
```diff
@@ -1577,16 +1582,19 @@ func Test_parseNetworkPolicyPeer(t *testing.T) {
 }
 
 func (s *K8sSuite) TestGetPolicyLabelsv1(c *C) {
+	uuid := "1bba160-ddca-11e8-b697-0800273b04ff"
 	tests := []struct {
 		np          *networkingv1.NetworkPolicy // input network policy
 		name        string                      // expected extracted name
 		namespace   string                      // expected extracted namespace
+		uuid        string                      //expected extracted uuid
```
Comment (aanm) [inline pkg/k8s/network_policy_test.go]: `//expected` -> `// expected`

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by eloycoto** [comment]
test-missed-k8s (netperf issue https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1201/testReport/junit/k8s-1/8/K8sChaosTest_Restart_with_long_lived_connections_L3_L4_policies_still_work_while_Cilium_is_restarted/)

**Comment by eloycoto** [comment]
test-missed-k8s

**Review by ianvernon — APPROVED** [review]
